hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java

```diff
@@ -380,12 +407,359 @@ public TssMessage getTssMessageFromBytes(Bytes wrap, TssParticipantDirectory dir
     }
 
     @Override
-    public void manageTssStatus(final State state) {
-        // TODO: Implement this method
+    public void manageTssStatus(
+            final State state,
+            final boolean isStakePeriodBoundary,
+            final Instant consensusNow,
+            final StoreMetricsService storeMetricsService) {
+        if (!appContext.configSupplier().get().getConfigData(TssConfig.class).keyCandidateRoster()) {
+            return;
+        }
+        final var readableStoreFactory = new ReadableStoreFactory(state);
+        final var tssStore = readableStoreFactory.getStore(ReadableTssStore.class);
+        final var rosterStore = readableStoreFactory.getStore(ReadableRosterStore.class);
+        // If the Tss Status is not computed yet during restart or reconnect, compute it from state.
+        if (tssStatus == null) {
+            tssStatus = computeInitialTssStatus(tssStore, rosterStore);
+        }
+
+        // collect tss encryption keys for all nodes in the active roster that are not null
```

Comment:
⚠️ In a keying stage, we use the encryption keys of the **target roster**, not the source roster. (Because members of the target roster are the ones that will be decrypting their private shares.)

So if we are keying the active roster, the below lookup is correct.

But if we are keying a candidate roster with lots of new/different members, looking up the encryption keys for the active roster could mislead us to think we have enough encryption keys available before we really do.

Reply:
Agreed. Changed to target roster. Thanks @tinker-michaelj